### DISCUSSION TITLE
Base64 URL-encoding

### DISCUSSION THREAD

**Discussion by dagleger2** [pkimetal/pkimetal#161] [Q&A] [answered]
Hi,
I have another question.
I have been testing the POST /linttbscert endpoint.
I could be wrong, but to me it looks like pkimetal expects the request to specify Content-Type: application/x-www-form-urlencoded, but at the same time actually expects the value of b64input to be _not_ URL-encoded?

(As a sidenote, are there any plans to have pkimetal accept Content-Type: application/json also?)

Regards
Dag

**Comment by dagleger2** [comment]
Hi again,
sorry, my mistake - the URL encoding was not the problem.
I eventually found out that my webclient-component sent Content-Type as "application/x-www-form-urlencoded;charset=UTF-8".
When I removed the "charset" part from the header value, the linttbs-request went through.

**Reply by robstradling** [reply, marked as answer]
Hi @dagleger2.

> Hi again,
> sorry, my mistake - the URL encoding was not the problem.

No need to apologise!  You've actually uncovered a problem...

> I eventually found out that my webclient-component sent Content-Type as "application/x-www-form-urlencoded;charset=UTF-8".
> When I removed the "charset" part from the header value, the linttbs-request went through.

The `charset` part is allowed according to [RFC7231 section 3.1.1.5](https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.1.5), so pkimetal really shouldn't complain if `charset` is present.

Fixed by https://github.com/pkimetal/pkimetal/pull/181.

> (As a sidenote, are there any plans to have pkimetal accept Content-Type: application/json also?)

No current plans, but if it's a feature you would use then please feel free to [add an issue](https://github.com/pkimetal/pkimetal/issues/new).
